### ISSUE TITLE
Option to save further outlinks of each url in Bulk save

### ISSUE THREAD

**Issue by Melonadev** [internetarchive/wayback-machine-webextension#691]
The extension itself already supports archiving outlinks of a webpage the user is visiting. So why not also do that for each url in Bulk save?

Example: [Madness Serial](https://madnessserial.com) is a website that offers free e-books, some of which are editions of existing public domain books.

Take this page for [Moby Dick](https://madnessserial.com/mdash/moby-dick-herman-melville). Here're the download formats:
![image](https://user-images.githubusercontent.com/66422959/101246243-0886e600-374d-11eb-9598-bfd2d5f0f7ab.png)

The trouble is that the [PDF reader](https://static1.squarespace.com/static/5ea988cb2f98c73c672bd6bb/t/5fb05d9d9a1aef48f68024e4/1605393892092/Moby+Dick%3B+or%2C+The+Whale+-+Herman+Melville.pdf) is not hosted on Madness Serial, but on a Squarespace domain. That's the reason why the [XML sitemap generator that I use](https://www.web-site-map.com/) doesn't include it.

That PDF is in just one of the non-Madness-Serial outlinks from the earlier Moby Dick page, and many of them are not in my list of urls. This is where this feature I'm suggesting comes in.

**Comment by cgorringe** [comment]
Thanks for the suggestion! Yes we can add the Outlinks and Screenshot options to Bulk Save. (although Outlinks will slow it down a bit.)

**Comment by Melonadev** [comment]
Exploring this idea further, let's say that the url list is like this:

**url 1** (outlinks are **url 2** & **url 3**)
**url 2** (outlink is **url 3**)
**url 3** (outlinks are **url 1** & **url 4**)
**url 4**
**url 5**

How about adding a _time-saving option_ (maybe make this the default?) where, instead of saving every single outlink for each url without checking it's been saved already in the same session, the extension does this:

1. Archive the first url, which is **url 1**, and detect and archive its outlinks **url 2** and **url 3**.
2. Proceed to the next url in the list, which is **url 2**.
3. Detect that **url 2** has already been archived earlier _this session_.
4. _Skip_ archiving **url 2**, but continue to detect its outlink **url 3** (also skipped because already archived earlier this session).
5. Proceed to the next url in the list, which is **url 3**.
6. _Skip_ **url 3**, but continue to detect its outlinks **url 1** and **url 4**.
7. _Skip_ **url 1**, but archive **url 4**.
8. Proceed to the next url in the list, which is **url 4**.
9. _Skip_ **url 4**, but continue to detect any outlinks (in this case, none).
10. Archive **url 5**, and continue to detect any outlinks (in this case, none).

**Comment by cgorringe** [comment]
Interesting idea!

Although the extension doesn't do the parsing out of the outlinks. It's done on the server side, which will return a list of outlinks when calling the Status API while it's processing a Save call. This means I can't just _skip_ saving a URL but then continue to detect its outlinks.

In the case where let's say **url2** was was saved previously as an outlink of **url1**, when the same URL is encountered in the list again, attempting to Save it will return a warning and will _likely_ be skipped if already saved within the last 10 minutes (by default). Which means **url2**'s outlinks would likely not be saved.

So basically: 1. I don't think I can implement the above. 2. The same URL already doesn't get saved again if it was saved within the past 10 min. 3. Some outlinks might not get saved. (not sure how frequent this case is in practice!)

**Comment by Yashraj1406** [comment]
@cgorringe please assign me this issue, I would like to work on this
